portal/src/graphql/adminapi/RolesScreen.tsx

```diff
@@ -64,36 +105,92 @@ const RolesScreen: React.VFC = function RolesScreen() {
     }
     return encodeOffsetToCursor(offset - 1);
   }, [isSearch, offset]);
-  const { data, loading } = useQuery<
+
+  const onChangeOffset = useCallback((offset) => {
+    setOffset(offset);
+  }, []);
+
+  const onChangeSearchKeyword = useCallback((_e, value) => {
+    if (value != null) {
+      setSearchKeyword(value);
+      // Reset offset when search keyword was changed.
+      setOffset(0);
+    }
+  }, []);
+
+  const onClearSearchKeyword = useCallback((_e) => {
+    setSearchKeyword("");
+    // Reset offset when search keyword was changed.
+    setOffset(0);
+  }, []);
+
+  const localSearchBoxProps: LocalSearchBoxProps = useMemo(() => {
+    return {
+      className: styles.searchBox,
+      placeholder: renderToString("search"),
+      value: searchKeyword,
+      onChange: onChangeSearchKeyword,
+      onClear: onClearSearchKeyword,
+    };
+  }, [
+    renderToString,
+    searchKeyword,
+    onChangeSearchKeyword,
+    onClearSearchKeyword,
+  ]);
+
+  const { data, loading, previousData } = useQuery<
     RolesListQueryQuery,
     RolesListQueryQueryVariables
   >(RolesListQueryDocument, {
     variables: {
-      searchKeyword,
-      pageSize,
+      pageSize: isSearch ? searchResultSize : pageSize,
+      searchKeyword: debouncedSearchKey,
       cursor,
     },
     fetchPolicy: "network-only",
   });
 
+  const isInitialLoading = loading && previousData == null;
+
   const items = useMemo(() => {
     return [{ to: ".", label: <FormattedMessage id="RolesScreen.title" /> }];
   }, []);
 
-  const isEmpty = !loading && data?.roles?.edges?.length === 0;
+  const isEmpty = !isInitialLoading && (data?.roles?.edges?.length ?? 0) === 0;
 
   return (
-    <ScreenContent className={styles.content} layout="list">
-      <div className={styles.widget}>
-        <NavBreadcrumb className="block" items={items} />
-      </div>
-      {isEmpty ? (
-        <RolesScreenEmptyState className={styles.widget} />
-      ) : (
-        // TODO: implement list view
-        <RolesScreenEmptyState className={styles.widget} />
-      )}
-    </ScreenContent>
+    <LocalSearchBoxContext.Provider value={localSearchBoxProps}>
```

Comment:
What is the reason we need this provider, instead of passing the props directly?

Reply:
And also the word `Local` make me think it is searching in local, but it seems not. The search is performed in server side.

Reply:
I copied it from `UsersScreen` :sosad:

Let me fix it together